src/input-states.js

```diff
@@ -208,7 +210,9 @@ function setPortStateBySocketId(node, vnid, portName, socketId, state) {
             node.vnis[vnid].inputStates = node.vnis[vnid].inputStates || {};
             node.vnis[vnid].inputStates[portName] = node.vnis[vnid].inputStates[portName] || [];
         }
-        node.vnis[vnid].inputStates[portName][socketId] = state;
+        // Overwrite existing state object
+        var states = node.vnis[vnid].inputStates[portName];
+        states[socketId] = _.extend(states[socketId] || {}, state);
```

Comment:
This is merging state properties rather than setting an entirely new state object, whereas the API was previously setting an entirely new state object.  I think will be confusing to have some setters merge and others set the entire object.  If a setter is going to merge then I think it would be clearer to give it a different name in the API, such as mergeState.